src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -58,195 +63,147 @@ Device and XCLBIN
 
-    - The ``xrt::device`` class's constructor is used to open the device
-    - The member function ``xrt::device::load_xclbin`` is used to load the XCLBIN from the filename. 
-    - The member function ``xrt::device::load_xclbin`` returns the XCLBIN UUID, which is required to open the kernel (refer the Kernel Section). 
 
+The ``xrt::device::get_info()`` is a useful member function to obtain necessary information about a device. Some of the information such as Name, BDF can be used to select a specific device to load an XCLBIN
+
+.. code:: c++
+      :number-lines: 10
+      
+           std::cout << "device name:     " << device.get_info<xrt::info::device::name>() << "\n";
+           std::cout << "device bdf:      " << device.get_info<xrt::info::device::bdf>() << "\n";
 
 Buffers
 -------
 
 Buffers are primarily used to transfer the data between the host and the device. The Buffer related APIs are discussed in the following three subsections
 
-       1. Buffer allocation and deallocation
-       2. Data transfer using Buffers
-       3. Miscellaneous other Buffer APIs
+1. Buffer allocation and deallocation
+2. Data transfer using Buffers
+3. Miscellaneous other Buffer APIs
 
 
 
 1. Buffer allocation and deallocation
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-XRT APIs provides API for
+The C++ interface for buffers as below
    
-      - ``xrtBOAlloc``: Allocates a buffer object 4K aligned, the API must be called with appropriate flags. 
-      - ``xrtBOAllocUserPtr``: Allocates a buffer object using pointer provided by the user. The user pointer must be aligned to 4K boundary. 
-      - ``xrtBOFree``: Deallocates the allocated buffer. 
+The class constructor ``xrt::bo`` is mainly used to allocates a buffer object 4K aligned. By default, a regular buffer is created (optionally the user can create other types of buffers by providing a flag). 
 
-.. code:: c
+.. code:: c++
       :number-lines: 15
            
-           xrtMemoryGroup bank_grp_idx_0 = xrtKernelArgGroupId(kernel, 0);
-           xrtMemoryGroup bank_grp_idx_1 = xrtKernelArgGroupId(kernel, 1);
-
-           xrtBufferHandle input_buffer = xrtBOAlloc(device, buffer_size_in_bytes, XRT_BO_FLAGS_NONE, bank_grp_idx_0);
-           xrtBufferHandle output_buffer = xrtBOAlloc(device, buffer_size_in_bytes, XRT_BO_FLAGS_NONE, bank_grp_idx_1);
+           auto bank_grp_idx_0 = kernel.group_id(0);
+           auto bank_grp_idx_1 = kernel.group_id(1);
+    
+           auto input_buffer = xrt::bo(device, buffer_size_in_bytes,bank_grp_idx_0);
+           auto output_buffer = xrt::bo(device, buffer_size_in_bytes, bank_grp_idx_1);
 
-           ....
-           ....
-           xrtBOFree(input_buffer);
-           xrtBOFree(output_buffer);
+In the above code ``xrt::bo`` buffer objects are created using the class's constructor. Please note the following 
 
-   
-The above code block shows 
+- As no special flags are used a regular buffer will be created.  
```

Comment:
What is a regular buffer?
It is a buffer that has a host backing pointer allocated by user space in heap memory and a device buffer allocated in the specified bank.

Reply:
added this note